rpcs3/Emu/Cell/Modules/cellRtc.cpp

```diff
@@ -3,298 +3,1672 @@
 
 #include "cellRtc.h"
 
+#include "Emu/Cell/lv2/sys_time.h"
+#include "Emu/Cell/lv2/sys_memory.h"
+#include "Emu/Cell/lv2/sys_ss.h"
+
+//#include <iomanip>
+//#include <sstream>
+
 LOG_CHANNEL(cellRtc);
 
+template <>
+void fmt_class_string<CellRtcError>::format(std::string& out, u64 arg)
+{
+	format_enum(out, arg, [](auto error)
+	{
+		switch (error)
+		{
+			STR_CASE(CELL_RTC_ERROR_NOT_INITIALIZED);
+			STR_CASE(CELL_RTC_ERROR_INVALID_POINTER);
+			STR_CASE(CELL_RTC_ERROR_INVALID_VALUE);
+			STR_CASE(CELL_RTC_ERROR_INVALID_ARG);
+			STR_CASE(CELL_RTC_ERROR_NOT_SUPPORTED);
+			STR_CASE(CELL_RTC_ERROR_NO_CLOCK);
+			STR_CASE(CELL_RTC_ERROR_BAD_PARSE);
+			STR_CASE(CELL_RTC_ERROR_INVALID_YEAR);
+			STR_CASE(CELL_RTC_ERROR_INVALID_MONTH);
+			STR_CASE(CELL_RTC_ERROR_INVALID_DAY);
+			STR_CASE(CELL_RTC_ERROR_INVALID_HOUR);
+			STR_CASE(CELL_RTC_ERROR_INVALID_MINUTE);
+			STR_CASE(CELL_RTC_ERROR_INVALID_SECOND);
+			STR_CASE(CELL_RTC_ERROR_INVALID_MICROSECOND);
+		}
+
+		return unknown;
+	});
+}
+
+// Grabbed from JPCSP
+// This is the # of microseconds between January 1, 0001 and January 1, 1970.
+const u64 RTC_MAGIC_OFFSET = 62135596800000000ULL;
+// This is the # of microseconds between January 1, 0001 and January 1, 1601 (for Win32 FILETIME.)
+const u64 RTC_FILETIME_OFFSET = 50491123200000000ULL;
+
+// 400 years is a convenient number, since leap days and everything cycle every 400 years.
+// 400 years is in other words 20871 full weeks.
+const u64 rtc400YearTicks = 20871ULL * 7ULL * 24ULL * 3600ULL * 1000000ULL;
```

Comment:
We usually only use yeetCase for the stuff that is used in cell documentation.
There's a lot of yeetCase going on in this file.